internal/settings/providers/allinkl/provider.go

```diff
@@ -0,0 +1,172 @@
+	username      string
+	password      string
+	useProviderIP bool
+}
+
+func New(data json.RawMessage, domain, host string,
+	ipVersion ipversion.IPVersion) (p *provider, err error) {
+	extraSettings := struct {
+		Username      string `json:"username"`
+		Password      string `json:"password"`
+		UseProviderIP bool   `json:"provider_ip"`
+	}{}
+	if err := json.Unmarshal(data, &extraSettings); err != nil {
+		return nil, err
+	}
+	p = &provider{
+		domain:        domain,
+		host:          host,
+		ipVersion:     ipVersion,
+		username:      extraSettings.Username,
+		password:      extraSettings.Password,
+		useProviderIP: extraSettings.UseProviderIP,
+	}
+	if err := p.isValid(); err != nil {
+		return nil, err
+	}
+	return p, nil
+}
+
+func (p *provider) isValid() error {
+	switch {
+	case len(p.username) == 0:
+		return errors.ErrEmptyUsername
+	case len(p.password) == 0:
+		return errors.ErrEmptyPassword
+	case p.host == "*":
+		return errors.ErrHostWildcard
+	}
+	return nil
+}
+
+func (p *provider) String() string {
+	return utils.ToString(p.domain, p.host, constants.AllInkl, p.ipVersion)
+}
+
+func (p *provider) Domain() string {
+	return p.domain
+}
+
+func (p *provider) Host() string {
+	return p.host
+}
+
+func (p *provider) IPVersion() ipversion.IPVersion {
+	return p.ipVersion
+}
+
+func (p *provider) Proxied() bool {
+	return false
+}
+
+func (p *provider) BuildDomainName() string {
+	return utils.BuildDomainName(p.host, p.domain)
+}
+
+func (p *provider) HTML() models.HTMLRow {
+	return models.HTMLRow{
+		Domain:    models.HTML(fmt.Sprintf("<a href=\"http://%s\">%s</a>", p.BuildDomainName(), p.BuildDomainName())),
+		Host:      models.HTML(p.Host()),
+		Provider:  "<a href=\"https://all-inkl.com/\">ALL-INKL.com</a>",
+		IPVersion: models.HTML(p.ipVersion.String()),
+	}
+}
+
+func (p *provider) Update(ctx context.Context, client *http.Client, ip net.IP) (newIP net.IP, err error) {
+	u := url.URL{
+		Scheme: "https",
+		Host:   "dyndns.kasserver.com",
+		Path:   "/",
+		User:   url.UserPassword(p.username, p.password),
+	}
+	values := url.Values{}
+	values.Set("host", utils.BuildURLQueryHostname(p.host, p.domain))
+	if !p.useProviderIP {
+		if ip.To4() == nil { // ipv6
+			values.Set("myip", ip.String())
+		} else {
+			values.Set("myip", ip.String())
+		}
+	}
+	u.RawQuery = values.Encode()
+
+	request, err := http.NewRequestWithContext(ctx, http.MethodGet, u.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+	headers.SetUserAgent(request)
+
+	response, err := client.Do(request)
+	if err != nil {
+		return nil, err
+	}
+	defer response.Body.Close()
+
+	b, err := io.ReadAll(response.Body)
+	if err != nil {
+		return nil, fmt.Errorf("%w: %s", errors.ErrUnmarshalResponse, err)
+	}
+	s := string(b)
+
+
+	if response.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("%w: %d: %s", errors.ErrBadHTTPStatus, response.StatusCode, s)
+	}
+
+	switch s {
+	case "":
+		return nil, errors.ErrNoResultReceived
+	case constants.Nineoneone:
```

Comment:
Additional codes taken from #306 but I didn't check the documentation

```suggestion
	case constants.Notfqdn:
		return nil, errors.ErrHostnameNotExists
	case strings.HasPrefix(str, "badrequest"):
		return nil, errors.ErrBadRequest
	case constants.Nineoneone:
```

Reply:
This is not applied